datafusion/sql/src/statement.rs

```diff
@@ -1588,6 +1592,37 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         self.statement_to_plan(rewrite.pop_front().unwrap()) // length of rewrite is 1
     }
 
+    fn show_functions_to_plan(
+        &self,
+        filter: Option<ShowStatementFilter>,
+    ) -> Result<LogicalPlan> {
+        if filter.is_some() {
```

Comment:
are we planning to apply the filter in future? 

For example for SHOW now it is possible to display specific parameter like 
```
show datafusion.execution.batch_size
```

I think it would be interesting to show the function by filter so ther user can get the name and ideally the signature. 

Later we can use this metadata table to obtain correct signature in unified way instead of hardcoding signatures like now

Reply:
that is a good idea.
however, i think sqlparser-rs has the syntax potentialy mixed up.
the filter syntax is borrowed from MySQL while the statement itself isn't borrowed from there, creating a unique mix. see https://github.com/sqlparser-rs/sqlparser-rs/issues/1399
it would be best to resolve this (either fix or close the issue) before jumping on the filtering support